### ISSUE TITLE
Integrate with GitHub Actions

### ISSUE THREAD

**Issue by rafavallina** [aws/aws-proton-public-roadmap#7]
Allow AWS Proton to create, manage and monitor pipelines in GitHub Actions, including parametrizing the GitHub Actions file and committing it to the service repository

**Comment by coreycoto** [comment]
Yes please. 

It would be great if there was a super simple way to set up a new GitHub repo using an AWS Proton Template which creates a CI/CD pipeline for my new GitHub repo and deploys to my AWS account using GitHub Secrets.

**Comment by rafavallina** [comment]
Good point! One thing that I didn't include here, but we would like to work on, is the capacity for Proton to create a repo with skeleton code, so that devs do not need to have their repo ready ahead of time. I'll create an issue!

**Comment by alwell-kevin** [comment]
@imjohnbo - Have you seen this?

cc: @nickismyname

**Comment by JohnPreston** [comment]
> Good point! One thing that I didn't include here, but we would like to work on, is the capacity for Proton to create a repo with skeleton code, so that devs do not need to have their repo ready ahead of time. I'll create an issue!

I am surprised there is not already a CFN Resource done like we have NewRelic etc. via CFN for Github to create repositories the same way we can do for AWS CodeCommit?

From there one could setup the AWS CodeStar connections etc. ?

Also if we are to do work to integrate better AWS and Github Actions, then can we put forward to request better integration with Github all together ?
ie. Codebuild already does report when builds are successful etc, but then if one uses report groups, for testing or coverage, it does not report that. Also CodeGuru scans etc. should definitely report findings in a similar way as SonarCloud etc. do.

Proton might be the reason we bring those things together but really I think it would be great for all the community to have a better AWS and Github integration for all things to do with code.

Also (for codepipeline to implement) would be validation that Lambda code has been signed ? (unrelated to Proton specifically, again)

**Comment by rafavallina** [comment]
> I am surprised there is not already a CFN Resource done like we have NewRelic etc. via CFN for Github to create repositories the same way we can do for AWS CodeCommit?

I'm have to double check if there is one, but what I was hinting at is that we'd like Proton templates to have a way to include skeleton code as well, so that Proton can create a new service that includes the architecture, the pipeline and the skeleton code as well.

I will direct PMs from other docs to read the rest of comments, as they pertain to services that we rely on as well :)

**Comment by marcusvoss** [comment]
I am not sure if there is a difference between github.com and enterprise server in the terms of this feature request, but i would be nice if you keep both variants of github in mind. Thanks
